hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/OzoneManagerProtocolClientSideTranslatorPB.java

```diff
@@ -1004,6 +1006,46 @@ public List<OmKeyInfo> listKeys(String volumeName, String bucketName,
 
   }
 
+  /**
+   * Lightweight listKeys implementation.
+   */
+  @Override
+  public List<BasicOmKeyInfo> listKeysLight(String volumeName,
+      String bucketName, String startKey, String prefix,
+      int maxKeys) throws IOException {
+    List<BasicOmKeyInfo> keys = new ArrayList<>();
+    ListKeysRequest.Builder reqBuilder = ListKeysRequest.newBuilder();
+    reqBuilder.setVolumeName(volumeName);
+    reqBuilder.setBucketName(bucketName);
+    reqBuilder.setCount(maxKeys);
+
+    if (startKey != null) {
```

Comment:
nit: should we add empty check too? If yes, `StringUtils.isNotEmpty()` or `StringUtils.isNotBlank()` can be used for that.

Reply:
Yes, thank you, I have added the empty check here.